### PR TITLE
Bootstrap script

### PR DESCRIPTION
**Pull request by aminya**

### CODE REVIEW

**Comment by joehowarth17** [comment]
So this seems to work!
I've was able to run the script from Powershell and WSL, and both result in the SoftConsole project compiling successfully.

When running from WSL, the arm-none-eabi toolchain must be installed before, because the softconsole toolchain only works from windows. We can have the script automatically download the toolchain if we want.

When I tried to run from powershell I was initially getting errors from the python waf build script. These seem to be related to the version of python, and versions < 3.7 should be okay.  With version 3.7.3 it does not work, and I can confirm that it works with version 3.6.5.

If we upgrade CSP, there is an updated wscript and version of waf that will probably work with newer versions of python.
But there is lots of changes (100+ commits) to the most recent CSP, so it is probably best to upgrade later and just use python 3.6 for now.

File: scripts/bootstrap.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+libraries_path = "./iris-fsw-softconsole/Libraries"
+
+################################################################
+# Environment variables
+import find_os
+import os.path
+
+if find_os.is_windows:
+    import glob
+    microchip_toolchain_glob = glob.glob("C:/Microsemi/*/arm-none-eabi-gcc/bin")
```
Comment (aminya) [inline scripts/bootstrap.py:39]: It is Microchip on my system. I think we should try both. 
```suggestion
    microchip_toolchain_glob = glob.glob("C:/Microchip/*/arm-none-eabi-gcc/bin") +  glob.glob("C:/Microsemi/*/arm-none-eabi-gcc/bin")
```

**Comment by aminya** [comment]
I get this error:
```py
❯ python .\scripts\bootstrap.py
Changing current working directory to the repository root
Downloading submodules...
Building libcsp...
configuring...
Setting top to                           : C:\Users\aminy\Documents\GitHub\Cpp\IrisSat-Flight-Software\iris-fsw-softconsole\Libraries\libcsp
Setting out to                           : C:\Users\aminy\Documents\GitHub\Cpp\IrisSat-Flight-Software\iris-fsw-softconsole\Libraries\libcsp\build
Checking for program 'gcc, cc'           : C:\Microchip\SoftConsole-v6.5\arm-none-eabi-gcc\bin\arm-none-eabi-gcc.exe
Checking for program 'ar'                : C:\Microchip\SoftConsole-v6.5\arm-none-eabi-gcc\bin\arm-none-eabi-ar.exe
The system cannot find the path specified.
Checking for endianness                  : no
The configuration failed
(complete log in C:\Users\aminy\Documents\GitHub\Cpp\IrisSat-Flight-Software\iris-fsw-softconsole\Libraries\libcsp\build\config.log)
Traceback (most recent call last):
  File ".\scripts\bootstrap.py", line 86, in <module>
    subprocess.run(f'python waf configure --toolchain=arm-none-eabi- --enable-if-can --with-os=freertos --cflags "{optimization_flags}" --includes="{INCLUDES}"', shell=True, check=True)
  File "C:\Python36\lib\subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'python waf configure --toolchain=arm-none-eabi- --enable-if-can --with-os=freertos --cflags "-O3" --includes="../../FreeRTOS,../../FreeRTOS/FreeRTOS-Source,../../FreeRTOS/FreeRTOS-Source/include,../../FreeRTOS/FreeRTOS-Source/portable,../../FreeRTOS/FreeRTOS-Source/portable/GCC/ARM_CM3,"' returned non-zero exit status 1.
```

**Comment by aminya** [comment]
I fixed it in https://github.com/IrisSat/libcsp/commit/af0e0754600442379132cdf07fadc3f2275c650a

**Comment by aminya** [comment]
> So this seems to work!
> I've was able to run the script from Powershell and WSL, and both result in the SoftConsole project compiling successfully.
> 
> When running from WSL, the arm-none-eabi toolchain must be installed before, because the softconsole toolchain only works from windows. We can have the script automatically download the toolchain if we want.
> 
> When I tried to run from powershell I was initially getting errors from the python waf build script. These seem to be related to the version of python, and versions < 3.7 should be okay. With version 3.7.3 it does not work, and I can confirm that it works with version 3.6.5.
> 
> If we upgrade CSP, there is an updated wscript and version of waf that will probably work with newer versions of python.
> But there is lots of changes (100+ commits) to the most recent CSP, so it is probably best to upgrade later and just use python 3.6 for now.

Test on Python 3.9. It works without issues. 

We should definitely update the libcsp later in a different PR. 

I will update the instructions and then merge this PR

**Comment by aminya** [comment]
We don't provide a build solution for the other files of the project without softconsole. It is safe to assume that the people have softconsole installed. So, I don't think providing a script for downloading the toolchain helps much now.

**Comment by joehowarth17** [comment]
> 
> 
> > So this seems to work!
> > I've was able to run the script from Powershell and WSL, and both result in the SoftConsole project compiling successfully.
> > When running from WSL, the arm-none-eabi toolchain must be installed before, because the softconsole toolchain only works from windows. We can have the script automatically download the toolchain if we want.
> > When I tried to run from powershell I was initially getting errors from the python waf build script. These seem to be related to the version of python, and versions < 3.7 should be okay. With version 3.7.3 it does not work, and I can confirm that it works with version 3.6.5.
> > If we upgrade CSP, there is an updated wscript and version of waf that will probably work with newer versions of python.
> > But there is lots of changes (100+ commits) to the most recent CSP, so it is probably best to upgrade later and just use python 3.6 for now.
> 
> Test on Python 3.9. It works without issues.
> 
> We should definitely update the libcsp later in a different PR.
> 
> I will update the instructions and then merge this PR

Weird I just tried upgrading to python 3.9.1 and now the script gives errors. 
I'll just stick to 3.6 for now, since we can both build without errors at this point.

**Comment by aminya** [comment]
Oh sorry. That was my fault. I have both installed but python 3.6 was on the PATH. Yes, it does not work with the latest version.

I will try to update the submodule so we don't have this problem anymore.

**Pull request merged**